src/Collection/Collection.php

```diff
@@ -20,26 +20,17 @@ class_exists(Pagination::class);
  */
 class Collection implements IteratorAggregate, Countable
 {
-    private $elements;
-    private $count;
-    private $pagination;
-
-    /**
-     * @var Microservice|null
-     */
-    private $microservice;
+    private ?Microservice $microservice = null;
 
     /**
      * @param list<T> $elements
      */
     public function __construct(
-        array $elements,
-        int $count,
-        ?Pagination $pagination = null
+        private array $elements,
+        private int $count,
+        private ?Pagination $pagination = null
```

Comment:
```suggestion
        private ?Pagination $pagination = null,
```
Same for toher occurrences, it eases the future diff